sycl/source/detail/online_compiler/ocloc_api.h

```diff
@@ -0,0 +1,110 @@
+//===------- ocloc_api.h --------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This file is copied from
+// https://github.com/intel/compute-runtime/blob/master/shared/offline_compiler/source/ocloc_api.h
+
+#include <cstdint>
+
+#ifndef OCLOC_MAKE_VERSION
+/// Generates ocloc API versions
+#define OCLOC_MAKE_VERSION(_major, _minor)                                     \
+  ((_major << 16) | (_minor & 0x0000ffff))
+#endif // OCLOC_MAKE_VERSION
+
+typedef enum _ocloc_version_t {
+  OCLOC_VERSION_1_0 = OCLOC_MAKE_VERSION(1, 0),     ///< version 1.0
+  OCLOC_VERSION_CURRENT = OCLOC_MAKE_VERSION(1, 0), ///< latest known version
+  OCLOC_VERSION_FORCE_UINT32 = 0x7fffffff
+} ocloc_version_t;
```

Comment:
Looking at this code, perhaps it could be modernized upstream too if it is C++ by replacing `OCLOC_MAKE_VERSION` by a `constexpr` function in some nice `namespace` and the `enum` a C++ one without the `typedef`.
But this would be outside of this PR.

Reply:
That could be done. The alternative to modernizing the file is keeping it as close to the original as possible to make it easier follow the changes in the original file.
I did only clang-format changes over the original.

Reply:
To give more context for others: this file is copied from the underlying Gen driver source base. See the [discussion ](https://github.com/intel/llvm/pull/2930/files#r546952703)above.